pkg/netdevice/netInfoProviders.go

```diff
@@ -72,11 +72,21 @@ func NewVhostNetInfoProvider() types.DeviceInfoProvider {
 /* DeviceInfoProvider Interface */
 
 func (rip *vhostNetInfoProvider) GetDeviceSpecs() []*pluginapi.DeviceSpec {
+	deviceSpec := make([]*pluginapi.DeviceSpec, 2)
+
 	if !VhostNetDeviceExist() {
 		glog.Errorf("GetDeviceSpecs(): /dev/vhost-net doesn't exist")
 		return nil
 	}
-	return GetVhostNetDeviceSpec()
+	deviceSpec[0] = GetVhostNetDeviceSpec()[0]
+
+	if !TunDeviceExist() {
```

Comment:
This means we now require /dev/net/tun to exist when `needVhostNet` is set to `true` in resource config.

is there no use-case for vhost-net without tun device ?

Reply:
Thanks for the comment!

That was my question I am not aware from our side of any use case that we use vhost until now.
Now we have a use-case (dpdk slow path/ vpp) there is a need to create a tap device and for that, the tun device should be available inside the pod.

The other solution is to have a new flag for the tun device but I will like to know if there is a use-case for only vhost-net mount if not I don't see a reason to create another flag to will be always equal to the vhost one

Reply:
With our use case, we also need /dev/net/tun, it is exported to via:
      volumes:                                                                                       
      - hostPath:                                                                                    
          path: /dev/net/tun                                                                         
          type: ""                                                                                   
        name: net

Reply:
@amorenoz I see you also participated in discussions on the original vhostNet support PR #228 
any thoughts on this ? do you see a case where we need vhost but not `/dev/net/tun` ?

Reply:
Not with upstream DPDK. The vhost-net (virtio-user) PMD creates a tap device and configures it as the backend of the vhost-net device.

Reply:
ack, thanks for the input. if all use-cases sift down to needing tun as well, to me it makes sense to provide it as a mount when needVhostNet is specified.

lets see if we get additional inputs, maybe raise this in one of the NPWG network resource mgmt meeting ?

Reply:
I will do it at the next meeting thanks everyone for the inputs!